src/inc/volatile.h

```diff
@@ -106,26 +106,43 @@
 #define VOLATILE_MEMORY_BARRIER()
 #endif // __GNUC__
 
+template<typename T>
+struct RemoveVolatile
+{
+   typedef T type;
+};
+
+template<typename T>
+struct RemoveVolatile<volatile T>
+{
+   typedef T type;
+};
+
+
 //
 // VolatileLoad loads a T from a pointer to T.  It is guaranteed that this load will not be optimized
 // away by the compiler, and that any operation that occurs after this load, in program order, will
 // not be moved before this load.  In general it is not guaranteed that the load will be atomic, though
 // this is the case for most aligned scalar data types.  If you need atomic loads or stores, you need
 // to consult the compiler and CPU manuals to find which circumstances allow atomicity.
 //
+// Starting at version 3.8, clang errors out on initializing of type int * to volatile int *. To fix this, we add two templates to cast away volatility
```

Comment:
@janvorli It would be good to move to Clang3.9 as the default for arm64.  Looks like it has been backported to Ubuntu 14.04 and 16.04 and it includes a prebuilt compatible lldb.